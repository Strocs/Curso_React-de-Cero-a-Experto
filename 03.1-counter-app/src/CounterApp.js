import PropTypes from 'prop-types'
import { useState } from 'react'

const CounterApp = ({ value }) => {
	const [counter, setCounter] = useState(value)

	//handleAdd
	//setCounter( (c) => c + 1 )
	const   handleAdd = () => setCounter(counter + 1),
		    handleSubtract = () => setCounter(counter - 1),
		    handleReset = () => setCounter(value)

	return (
		<>
			<h1>CounterApp:</h1>
			<h2>{counter}</h2>

			<button onClick={ handleAdd }>+1</button>
			<button onClick={ handleReset }>Reset</button>
			<button onClick={ handleSubtract }>-1</button>
		</>
	)
}

CounterApp.propTypes = {
	value: PropTypes.number.isRequired,
}
CounterApp.defaultProps = {
	value: 0,
}

export default CounterApp
