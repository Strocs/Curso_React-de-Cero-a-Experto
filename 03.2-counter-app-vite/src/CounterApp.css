#root {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  text-align: center;
}

h1 {
  letter-spacing: .4rem;
  background: linear-gradient(90deg, cyan, blueviolet);
  background-clip: text;
  -webkit-background-clip: text;
  color: transparent;
}

h2 {
  color:lightskyblue;
  font-size: 3rem;
}
button {
  background: aliceblue;
  color: black;
  margin: 1rem;
}
