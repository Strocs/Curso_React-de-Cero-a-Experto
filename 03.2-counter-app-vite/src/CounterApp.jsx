import React, { useState } from 'react'
import PropTypes from 'prop-types'
import './CounterApp.css'

const CounterApp = ({ value = 0 }) => {
  const [count, setCount] = useState(value)

  return (
    <>
      <h1>Counter App</h1>
      <h2>{count}</h2>
      <div>
        <button onClick={() => setCount(count + 1)} type='button'>
          +1
        </button>
        <button onClick={() => setCount(count - 1)} type='button'>
          -1
        </button>
        <button
          aria-label='btn-reset'
          onClick={() => setCount(value)}
          type='button'
        >
          Reset
        </button>
      </div>
    </>
  )
}

export default CounterApp

CounterApp.propTypes = {
  value: PropTypes.number
}
