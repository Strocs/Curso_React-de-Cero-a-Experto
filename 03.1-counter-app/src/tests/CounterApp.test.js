import CounterApp from "../CounterApp"
import { shallow } from 'enzyme'


describe('Tests del CounterApp', () => { 
    //scope
    // se repite para no perder el intellisense
    let wrapper = shallow(<CounterApp />)
    const counter = wrapper.find('h2').text()
    // reinicializar el valor en cada test
    beforeEach(() => {
        wrapper = shallow(<CounterApp />)
    })

    test('should display <CounterApp /> correctly', () => { 
        expect(wrapper).toMatchSnapshot()
    })

    test('should display default valor of 100', () => { 
        const value = 100
        const wrapper = shallow(<CounterApp value={value} />)

        const counter = wrapper.find('h2').text()
        expect(counter).toBe('100')
    })

    test('should increment with +1 button', () => { 

        wrapper.find('button').at(0).simulate('click')
        // console.log(btn1.html()); // para ver el html
        const counter = wrapper.find('h2').text()
        expect(counter).toBe('1')

    })

    test('should increment with -1 button', () => { 
        //recibe el valor aumentado de la prueba anterior
        wrapper.find('button').at(2).simulate('click')
        // console.log(btn1.html()); // para ver el html
        const counter = wrapper.find('h2').text()
        expect(counter).toBe('-1')
    })

    test('should reset to default value = 0', () => { 

        wrapper.find('button').at(2).simulate('click')
        wrapper.find('button').at(1).simulate('click')
        const counter = wrapper.find('h2').text()
        expect(counter).toBe('0')
    })

 })