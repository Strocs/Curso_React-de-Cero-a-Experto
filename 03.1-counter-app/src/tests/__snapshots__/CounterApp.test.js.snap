// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests del CounterApp should display <CounterApp /> correctly 1`] = `
<Fragment>
  <h1>
    CounterApp:
  </h1>
  <h2>
    0
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
